public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -405,7 +404,7 @@ function getAnnotations(state: DashboardSceneState): AnnotationQueryKind[] {
         },
         enable: Boolean(layer.state.isEnabled),
         hide: Boolean(layer.state.isHidden),
-        filter: layer.state.query.filter ?? defaultAnnotationPanelFilter(),
+        filter: layer.state.query.filter,
```

Comment:
When updating snapshot for this and when filter is undefined this is persisted in the snapshot as `filter: undefined`, so maybe we should do something like: 
```
 if (layer.state.query.filter !== undefined) {
      result.spec.filter = layer.state.query.filter;
    }
```
Not sure if there's something in place on the backend or the frontend that cleans up these undefined fields and doesn't persist them in db.